backend/tests/unit/core/schema_manager/test_manager_schema.py

```diff
@@ -1963,6 +1964,62 @@ async def test_schema_branch_validate_check_missing(
     }
 
 
+async def test_schema_branch_validate_node_deletion(
+    db: InfrahubDatabase, reset_registry, default_branch: Branch, register_internal_models_schema
+):
+    FULL_SCHEMA = {
+        "nodes": [
+            {
+                "name": "Criticality",
+                "namespace": "Builtin",
+                "default_filter": "name__value",
+                "label": "Criticality",
+                "attributes": [
+                    {"name": "name", "kind": "Text", "label": "Name", "unique": True},
+                    {"name": "level", "kind": "Number", "label": "Level"},
+                    {"name": "color", "kind": "Text", "label": "Color", "default_value": "#444444"},
+                    {"name": "description", "kind": "Text", "label": "Description", "optional": True},
+                ],
+                "relationships": [
+                    {
+                        "name": "tags",
+                        "peer": InfrahubKind.TAG,
+                        "label": "Tags",
+                        "optional": True,
+                        "cardinality": "many",
+                    }
+                ],
+            },
+            {
+                "name": "Tag",
+                "namespace": "Builtin",
+                "label": "Tag",
+                "default_filter": "name__value",
+                "attributes": [
+                    {"name": "name", "kind": "Text", "label": "Name", "unique": True},
+                    {"name": "description", "kind": "Text", "label": "Description", "optional": True},
+                ],
+            },
+        ]
+    }
+    schema = SchemaRoot(**FULL_SCHEMA)
+    schema.generate_uuid()
+    schema_branch = SchemaBranch(cache={}, name="test")
+    schema_branch.load_schema(schema=schema)
+
+    FULL_SCHEMA["nodes"].pop(1)
+
+    broken_schema = SchemaRoot(**FULL_SCHEMA)
+    broken_schema_branch = SchemaBranch(cache={}, name="test-broken")
+    broken_schema_branch.load_schema(schema=broken_schema)
+
+    diff = schema_branch.diff(other=broken_schema_branch)
+    assert "BuiltinTag" in diff.removed
+
+    with pytest.raises(ValueError):
```

Comment:
would be good to check the exception message to make sure it is the expected one